gobblin-data-management/src/main/java/org/apache/gobblin/data/management/source/LoopingDatasetFinderSource.java

```diff
@@ -89,58 +89,70 @@ public LoopingDatasetFinderSource(boolean drilldownIntoPartitions) {
   }
 
   /**
-   * Collect all failed workunit in previous run into a set.
-   *
-   * Workunits that exceed max retry limit will not be added into failedPreviousWorkUnits
-   * and rollovered into next quota-available execution.
-   *
-   * @param retriedUrns Track all dataset/partition's urns that is included in the retried workunits, so that in the
-   *                    case where the number of remaining dataset/partitions are less than the maximum number of
-   *                    workunits can be processed by single execution, a retried workunit won't be processed twice
+   * Collect all failed workunit in previous run into a list.
+   * <p>
+   * Workunits that exceed max retry limit will not be added into failedPreviousWorkUnits and rolled over into next
+   * quota-available execution.
    */
-  private List<WorkUnit> getPreviousFailedWorkUnits(List<WorkUnitState> previousWorkUnitStates, int maxRetries,
-      Set<String> retriedUrns) {
+  private List<WorkUnit> getPreviousFailedWorkUnits(List<WorkUnitState> previousWorkUnitStates, int maxRetries) {
     List<WorkUnit> failedPreviousWorkUnits = new ArrayList<>();
     for (WorkUnitState workUnitState : previousWorkUnitStates) {
-      if ((workUnitState.getWorkingState() == WorkUnitState.WorkingState.FAILED
-          || workUnitState.getWorkingState() == WorkUnitState.WorkingState.ROLLOVER)) {
-        if (!workUnitState.contains(ConfigurationKeys.TASK_RETRIES_KEY)
-            || workUnitState.getPropAsInt(ConfigurationKeys.TASK_RETRIES_KEY) < maxRetries + 1) {
-          log.info("Dataset " + workUnitState.getProp(DATASET_URN) + " is failed previously, retrying...");
+      WorkUnitState.WorkingState workingState = workUnitState.getWorkingState();
+      if (workingState == WorkUnitState.WorkingState.FAILED || workingState == WorkUnitState.WorkingState.ROLLOVER) {
+        int currentRetryCount = 0;
+        if (workUnitState.contains(ConfigurationKeys.TASK_RETRIES_KEY)) {
+          currentRetryCount = workUnitState.getPropAsInt(ConfigurationKeys.TASK_RETRIES_KEY);
+        }
+        if (currentRetryCount < maxRetries) {
+          log.info("Dataset " + workUnitState.getProp(DATASET_URN) + " is failed previously, retrying for the "
+              + currentRetryCount + " times ");
```

Comment:
This log message doesn't make much sense if `currentRetryCount` is `0`, which it might be if `TASK_RETRIES_KEY` is not set in the `workUnitState`.